backend/src/bundles/users/user.service.ts

```diff
@@ -82,6 +84,60 @@ class UserService implements Service {
     public delete(): ReturnType<Service['delete']> {
         return Promise.resolve(true);
     }
+
+    public async addFollowing(
+        id: number,
+        followingId: number,
+    ): Promise<UserFollowingsResponseDto | null> {
+        try {
+            const addedFriend = await this.userRepository.addFollowing(
+                id,
+                followingId,
+            );
+            if (!addedFriend) {
+                throw new HttpError({
+                    message: UserValidationMessage.USER_NOT_FOUND,
+                    status: HttpCode.NOT_FOUND,
+                });
+            }
+            return addedFriend;
+        } catch (error) {
+            throw new Error(`Error occurred ${error}`);
+        }
+    }
+
+    public async removeFollowing(
+        id: number,
+        followingId: number,
+    ): Promise<number> {
+        try {
+            return await this.userRepository.removeFollowing(id, followingId);
+        } catch (error) {
+            throw new Error(`Error occurred while removing friend: ${error}`);
+        }
+    }
+
+    public async getFollowings(
+        userId: number,
+    ): Promise<UserFollowingsResponseDto[] | null> {
+        try {
+            return await this.userRepository.getFollowings(userId);
+        } catch (error) {
+            throw new Error(`Error occurred while fetching friends: ${error}`);
+        }
+    }
+
+    public async getNotFollowed(
+        userId: number,
+    ): Promise<UserFollowingsResponseDto[] | null> {
+        try {
+            return await this.userRepository.getNotFollowed(userId);
+        } catch (error) {
+            throw new Error(
+                `Error occurred while fetching not friends: ${error}`,
+            );
+        }
```

Comment:
I think you shouldn't use all of these repeating errors here as we have middleware for this.

Reply:
Thanks, I fixed it